### PR TITLE
feat: adding very basic confirmation page hidden behind env var

### LINKED ISSUE #11438: Create very basic re-designed confirmation root page page hidden behind env variable
Create very basic page for confirmations. As we start working on other confirmation page components like title, footer we can add to this page and it will help us figure state management details also.

Image

Figma: https://www.figma.com/design/CeAIcptzeaI5ciTpYBaiqr/Confirmations-DS?node-id=4314-18588&node-type=section&t=txw0EWWsBaXZnlBj-0

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Adding a very basic confirmation page hidden behind env flag.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/11438

## **Manual testing steps**
NA

## **Screenshots/Recordings**

https://github.com/user-attachments/assets/a8c702d1-bfd9-4613-a45a-b99c0e229b7b

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/confirmations/hooks/useRedesignEnabled.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { useMemo } from 'react';
+
+import { ApprovalTypes } from '../../../../core/RPCMethods/RPCMethodMiddleware';
+import useApprovalRequest from '../hooks/useApprovalRequest';
+
+const useRedesignEnabled = () => {
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/hooks/useRedesignEnabled.ts:6]: Should we be explicit here and call this `useConfirmationRedesignEnabled`?

**Review by pedronfigueiredo — APPROVED** [review]

File: app/components/Views/confirmations/hooks/useRedesignEnabled.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const { approvalRequest } = useApprovalRequest();
+  const approvalRequestType = approvalRequest?.type;
+
+  const isRedesignedEnabled = useMemo(
+    () =>
+      approvalRequestType &&
+      process.env.REDESIGNED_SIGNATURE_REQUEST === 'true' &&
+      approvalRequestType === ApprovalTypes.PERSONAL_SIGN,
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/hooks/useRedesignEnabled.ts:14]: Are we merging two concepts here, whether it's "enabled" due to the ENV and where a suitable approval request is present?

Could we instead just use `useApprovalRequest` and a personal sign type check within `Confirm` for now as that will eventually evolve into the `Info` component?

File: app/components/Views/confirmations/Confirm/Confirm.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  if (!isRedesignedEnabled) {
+    return null;
+  }
+
+  const styles = createStyles(colors);
+
+  return (
+    <BottomModal>
+      <View style={styles.container}>
+        <Text>TODO</Text>
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/Confirm/Confirm.tsx:22]: To verify this is rendering in the right conditions, is it worth rendering something from the approval request data?

File: app/components/Views/confirmations/Confirm/Confirm.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  const { isRedesignedEnabled } = useRedesignEnabled();
+
+  if (!isRedesignedEnabled) {
+    return null;
+  }
+
+  const styles = createStyles(colors);
+
+  return (
+    <BottomModal>
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/Confirm/Confirm.tsx:20]: Am I right this confirmation can't yet be approved or rejected?

Is it worth at least adding the unmount listener to reject it so this test confirmation can be actioned?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: app/components/Views/confirmations/Confirm/Confirm.tsx (hunk 4 above)
Comment (jpuri) [reply app/components/Views/confirmations/Confirm/Confirm.tsx:20]: I am working on next PR to add footer and accept / reject buttons.

File: app/components/Views/confirmations/Confirm/Confirm.tsx (hunk 3 above)
Comment (jpuri) [reply app/components/Views/confirmations/Confirm/Confirm.tsx:22]: More PR in queue for this.

File: app/components/Views/confirmations/hooks/useRedesignEnabled.ts (hunk 2 above)
Comment (jpuri) [reply app/components/Views/confirmations/hooks/useRedesignEnabled.ts:14]: This hook will go away as eventually all confirmations are re-designed.

File: app/components/Views/confirmations/hooks/useRedesignEnabled.ts (hunk 1 above)
Comment (jpuri) [reply app/components/Views/confirmations/hooks/useRedesignEnabled.ts:6]: I can update name in following PR.

File: app/components/Views/confirmations/Confirm/Confirm.tsx (hunk 4 above)
Comment (jpuri) [reply app/components/Views/confirmations/Confirm/Confirm.tsx:20]: https://github.com/MetaMask/metamask-mobile/issues/11434

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**